src/gui/guiutility.cpp

```diff
@@ -85,3 +93,32 @@ QString Utility::vfsFreeSpaceActionText()
 {
     return QCoreApplication::translate("utility", "Free up local space");
 }
+
+void Utility::setDirectoryTag(const QString &path)
+{
+    auto result = FileSystem::Tags::set(path, dirTag(), Theme::instance()->orgDomainName().toUtf8());
+    if (!result) {
+        qCWarning(lcGuiUtility) << QStringLiteral("Failed to set tag on '%1': %2").arg(path, result.error())
+#ifdef Q_OS_WIN
+                                << QStringLiteral("(filesystem %1)").arg(FileSystem::fileSystemForPath(path))
```

Comment:
Wouldn't that make more sense to check in FileSystem::Tags::set(? sry for not clarifying earlier

Reply:
I kinda like doing the handling of the error outside of the "low level" function.